### ISSUE TITLE
I cant build the iOS app after upgrading to Xcode 12

### ISSUE THREAD

**Issue by zumaawad** [Leanplum/Leanplum-ReactNative-SDK#23]
I cant build the iOS app after upgrading to Xcode 12
It returns

<img width="1166" alt="Screen Shot 2020-09-22 at 9 04 38 PM" src="https://user-images.githubusercontent.com/11634587/93926172-7b202200-fd17-11ea-8a78-92b69c4bbe94.png">

I'm using the latest of this repository with Xcode 12.0 (12A7209)
`"@leanplum/react-native-sdk": "https://github.com/Leanplum/Leanplum-ReactNative-SDK",`
`"react": "16.11.0",`
`"react-native": "0.62.2"`

**Comment by nzagorchev** [comment]
Hello,
Can you try the following:

- Delete Pod folder and Pod.lock file
- run pod repo update
- run pod install

I am able to compile the following app successfully with Xcode12:
https://github.com/Leanplum/Rondo-ReactNative

**Comment by zumaawad** [comment]
@nzagorchev Which `"@leanplum/react-native-sdk"` version?

**Comment by defualt** [comment]
![image](https://user-images.githubusercontent.com/399135/94470689-037a4900-0196-11eb-844f-b149fb27eb8b.png)

**Comment by nzagorchev** [comment]
Using version `1.1.0-beta5`

**Comment by defualt** [comment]
1.1.0-beta11


```
▸ Compiling Leanplum.swift

❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/LPVar.h:113:1: 'LPVar' with definition in module 'Leanplum' has different definitions in different modules; first difference is this method

- (NSString *)name;
                                                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/Leanplum.h:463:1: 'Leanplum' with definition in module 'react_native_leanplum' has different definitions in different modules; first difference is this method

+ (void)handleActionWithIdentifier:(NSString *)identifier
^



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/Leanplum.swift:119:40: type 'Leanplum' has no member 'LocationAccuracyType'

        if let accuracyType = Leanplum.LocationAccuracyType(rawValue: type) {
^


▸ Compiling LeanplumInbox.swift

❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/Leanplum.h:463:1: 'Leanplum' with definition in module 'react_native_leanplum' has different definitions in different modules; first difference is this method

+ (void)handleActionWithIdentifier:(NSString *)identifier
                              ~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~


▸ Compiling LeanplumTypeUtils.swift

❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/LPVar.h:113:1: 'LPVar' with definition in module 'Leanplum' has different definitions in different modules; first difference is this method

- (NSString *)name;
^
```

**Comment by defualt** [comment]
I just upgraded from React Native 0.62.2 to 0.63.3 but that actually churned up more error
```
▸ Compiling Leanplum.swift

❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/LPVar.h:113:1: 'LPVar' with definition in module 'Leanplum' has different definitions in different modules; first difference is this method

- (NSString *)name;
                           ^



❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/Leanplum.h:463:1: 'Leanplum' with definition in module 'react_native_leanplum' has different definitions in different modules; first difference is this method

+ (void)handleActionWithIdentifier:(NSString *)identifier
^



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/Leanplum.swift:119:40: type 'Leanplum' has no member 'LocationAccuracyType'

        if let accuracyType = Leanplum.LocationAccuracyType(rawValue: type) {
^



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/Leanplum.swift:198:24: argument passed to call that takes no arguments

        let lpVar = Var(name: name, file: filename)
                              ~~~~~~~~ ^~~~~~~~~~~~~~~~~~~~


▸ Compiling LeanplumInbox.swift

❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/Leanplum.h:463:1: 'Leanplum' with definition in module 'react_native_leanplum' has different definitions in different modules; first difference is this method

+ (void)handleActionWithIdentifier:(NSString *)identifier
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~


▸ Compiling LeanplumTypeUtils.swift

❌  /Users/my-app/ios/Pods/Leanplum-iOS-SDK/Leanplum-SDK/Classes/LPVar.h:113:1: 'LPVar' with definition in module 'Leanplum' has different definitions in different modules; first difference is this method

- (NSString *)name;
^



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:17:24: argument passed to call that takes no arguments

            lpVar = Var(name: key, double: value as? Double ?? 0.0)
^



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:19:24: argument passed to call that takes no arguments

            lpVar = Var(name: key, boolean: value as? Bool ?? false)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:21:24: argument passed to call that takes no arguments

            lpVar = Var(name: key, string: value as? String)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:23:24: argument passed to call that takes no arguments

            lpVar = Var(name: key, array: value as? Array)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:23:53: generic parameter 'Element' could not be inferred in cast to 'Array'

            lpVar = Var(name: key, array: value as? Array)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:25:24: argument passed to call that takes no arguments

            lpVar = Var(name: key, dictionary: value as? Dictionary)
                                                    ^



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:25:58: generic parameter 'Key' could not be inferred in cast to 'Dictionary'

            lpVar = Var(name: key, dictionary: value as? Dictionary)
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



❌  /Users/my-app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift:25:58: generic parameter 'Value' could not be inferred in cast to 'Dictionary'

            lpVar = Var(name: key, dictionary: value as? Dictionary)
                                                         ^

```

**Comment by milos1290** [comment]
@defualt Can you make sure your pods are updated. Go to ios folder and run 'pod repo update' and 'pod install' to pull the latest version of native iOS Leanplum-iOS SDK

**Comment by defualt** [comment]
I'm still getting errors.
I installed `Leanplum/Leanplum-ReactNative-SDK` version 1.1.0.
react-native version is 0.63.3.
Here are the commands I ran after npm installing the latest leanplum/react-native-sdk.
```
rm -rf node_modueles
rm -rf ios/Pods
rm ios/Podfile.lock
npm install
(cd ios && pod repo update && pod install)
```

The `Podfile.lock` file has this: 

```
  - Leanplum-iOS-SDK (3.0.0)

...

- react-native-leanplum (1.1.0):
    - Leanplum-iOS-SDK (= 3.0.0)
    - React
```

I ran "Clean Build Folder" from Xcode menus.
I then pressed play in Xcode.

I get these errors with a failed build: 

![image](https://user-images.githubusercontent.com/399135/94954860-b0670580-04b7-11eb-9d52-0325dbdd66e2.png)

Then when I try to run `react-native run-ios` I get these errors:

```
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal x86_64 /Users/app/ios/Pods/FBSDKShareKit/FBSDKShareKit/FBSDKShareKit/Swift/Enums+Extensions.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
	CompileSwift normal x86_64 /Users/app/node_modules/@leanplum/react-native-sdk/ios/Leanplum.swift
	CompileSwift normal x86_64 /Users/app/node_modules/@leanplum/react-native-sdk/ios/LeanplumInbox.swift
	CompileSwift normal x86_64 /Users/app/node_modules/@leanplum/react-native-sdk/ios/LeanplumTypeUtils.swift
	CompileSwift normal x86_64 /Users/app/node_modules/lottie-react-native/src/ios/LottieReactNative/AnimationViewManagerModule.swift
	CompileSwift normal x86_64 /Users/app/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
(7 failures)
```


Then when I try to Play again from Xcode I get these errors:

![image](https://user-images.githubusercontent.com/399135/94955905-4a7b7d80-04b9-11eb-8dca-3b2b30caf0f6.png)



When I uninstall `Leanplum/Leanplum-ReactNative-SDK` I get no errors and the build succeeds.

**Comment by zumaawad** [comment]
Any updates?

**Comment by defualt** [comment]
I hope to see progress on this issue because this is blocking a new release.   Let me know if you need any more information.

**Comment by jgalianoz** [comment]
Any updates ?

**Comment by jgalianoz** [comment]
@defualt @zumaawad  Do you compile the project in Legacy mode?

![image](https://user-images.githubusercontent.com/12010942/96615743-52179080-12c7-11eb-86cb-4ecc58c363b9.png)

When I compile in Legacy mode I receive this error:

![image](https://user-images.githubusercontent.com/12010942/96617021-b4bd5c00-12c8-11eb-8001-bcb2fa550290.png)


When I don't compile in Legacy mode I receive this error:

![image](https://user-images.githubusercontent.com/12010942/96615607-1b417a80-12c7-11eb-9744-b0fb1f1125f3.png)

**Comment by milos1290** [comment]
So what i did is this
```
yarn upgrade @leanplum/react-native-sdk OR npm update @leanplum/react-native-sdk
cd ios && pod update
```

And i was able to build iOS App from XCode

**Comment by defualt** [comment]
@milos1290 Please review my message [above](https://github.com/Leanplum/Leanplum-ReactNative-SDK/issues/23#issuecomment-702882728) where I confirm the versions installed with a fresh node_modules and Pods folders:
- `@leanplum/react-native-sdk": "1.1.0"`
- `Leanplum-iOS-SDK (3.0.0)`

Have you tried installing `lottie-react-native` along with `@leanplum/react-native-sdk`?  Lottie is mentioned in the errors, but only when Leanplum is also installed.

**Comment by milos1290** [comment]
Disregard my previous reply, i've managed to track the issue. 
Recently RN switched from using dynamic frameworks to static ones due to few libraries that are built as static (mainly Flipper). 

The only solution is to add 
`use_frameworks!`

 and remove 

```
  use_flipper!
  post_install do |installer|
    flipper_post_install(installer)
  end
```

Clean the project and it should build.

**Comment by jgalianoz** [comment]
@milos1290 I receive this error when removing this.

![image](https://user-images.githubusercontent.com/12010942/96658616-97a97d00-130a-11eb-8b2b-9a56b3ec50db.png)

**Comment by defualt** [comment]
@milos1290 I assume that disables Flipper.  Is there the possibility of another fix that keeps Flipper integration intact?

**Comment by nzagorchev** [comment]
We have also pushed a fix for module definition in the iOS SDK: [PR](https://github.com/Leanplum/Leanplum-iOS-SDK/pull/405)
I will be doing a beta soon which will include it.

**Comment by milos1290** [comment]
@jgalianoz @defualt I'm afraid there is nothing we can do atm. I'm still trying to find other solution to would allow us to link SDK dynamically and successfully build the app

**Comment by milos1290** [comment]
@defualt @jgalianoz I'm released a new version for RN SDK, it will contain a partial fix for XCode12 and RN 0.63 issues. I'll post a guide on how to proceed

**Comment by milos1290** [comment]
Please refer to this issue or a fix https://github.com/Leanplum/Leanplum-ReactNative-SDK/issues/30

**Comment by nzagorchev** [comment]
> Please refer to this issue or a fix #30
